### PR TITLE
Add GreatPad macropad under gkeyboard

### PR DESCRIPTION
**Pull request by gkeyboard**
Add GreatPad macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gkeyboard/greatpad/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#define OLED_BRIGHTNESS 128
+#define OLED_FONT_H  "lib/glcdfont.c"
+
+#define ENCODER_DIRECTION_FLIP          // If your encoder’s clockwise directions are incorrect, you can swap the A & B pad definitions
+
+#define RGB_MATRIX_LED_COUNT 28
+
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES           // reacts to keypresses
+#define RGB_DISABLE_WHEN_USB_SUSPENDED  // turn off effects when suspended
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/config.h:19]:
```suggestion
#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
#define RGB_MATRIX_KEYPRESSES
```

File: keyboards/gkeyboard/greatpad/greatpad.c [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+        case 9:
+            render_LAYER_9();
+            break;
+        default:
+            render_LAYER_UNDEF();
+    }
+}
+
+#define SHOW_LOGO 5000
+bool oled_task_user(void) {
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/greatpad.c:561]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
```

File: keyboards/gkeyboard/greatpad/info.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "layer_count": 10
+    },
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            {"pin_a": "GP26", "pin_b": "GP25"},
+            {"pin_a": "GP1", "pin_b": "GP0"}
+        ]
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/info.json:36]:
```suggestion
        "rotary": [
            {"pin_a": "GP25", "pin_b": "GP26"},
            {"pin_a": "GP0", "pin_b": "GP1"}
        ]
```

File: keyboards/gkeyboard/greatpad/info.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
+        "max_brightness": 170,
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/info.json:40]:
```suggestion
        "max_brightness": 170,
        "sleep": true,
```

File: keyboards/gkeyboard/greatpad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    TO(9)
+    ),
+    [9] = LAYOUT(
+        KC_TRNS,                                    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    TO(0)
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/keymaps/default/keymap.c:76]: Blank layers should be removed.
```suggestion
        KC_P0,    KC_PDOT,  KC_PENT,    KC_PPLS,    _______
```

File: keyboards/gkeyboard/greatpad/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),  ENCODER_CCW_CW(RGB_RMOD, RGB_MOD) },
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [4] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [5] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [6] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [7] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [8] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [9] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/keymaps/default/keymap.c:91]:
```suggestion
```

File: keyboards/gkeyboard/greatpad/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    TO(9)
+    ),
+    [9] = LAYOUT(
+        KC_TRNS,                                    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,
+        KC_TRNS,  KC_TRNS,  KC_TRNS,    KC_TRNS,    TO(0)
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/keymaps/via/keymap.c:76]: Blank layers should be removed.
```suggestion
        KC_P0,    KC_PDOT,  KC_PENT,    KC_PPLS,    _______
```

File: keyboards/gkeyboard/greatpad/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),  ENCODER_CCW_CW(RGB_RMOD, RGB_MOD) },
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [4] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [5] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [6] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [7] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [8] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [9] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),  ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/keymaps/via/keymap.c:91]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/gkeyboard/greatpad/lib/glcdfont.c
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/lib/glcdfont.c:1]: Please add a license header to this file. eg.
```
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Comment (gkeyboard) [reply keyboards/gkeyboard/greatpad/lib/glcdfont.c:1]: Thanks you.

File: .vscode/extensions.json
Comment (fauxpark) [inline .vscode/extensions.json:1]: Please restore this file.

File: .vscode/settings.json
Comment (fauxpark) [inline .vscode/settings.json:1]: Please restore this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: .vscode/extensions.json
Comment (gkeyboard) [reply .vscode/extensions.json:1]: OK

File: .vscode/settings.json
Comment (gkeyboard) [reply .vscode/settings.json:1]: OK

File: keyboards/gkeyboard/greatpad/mcuconf.h [hunk 9]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2023 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef RP_I2C_USE_I2C0
+#undef RP_I2C_USE_I2C1
+#define RP_I2C_USE_I2C0 TRUE
+#define RP_I2C_USE_I2C1 FALSE
```
Comment (waffle87) [inline keyboards/gkeyboard/greatpad/mcuconf.h:10]:
```suggestion
#undef RP_I2C_USE_I2C0
#define RP_I2C_USE_I2C0 TRUE
```

File: .vscode/settings.json [hunk 10]
```diff
@@ -30,5 +30,6 @@
     },
     "clangd.arguments": [
         "--header-insertion=never"
-    ]
+    ],
+    "cmake.sourceDirectory": "C:/qmk_firmware/lib/googletest"
```
Comment (fauxpark) [inline .vscode/settings.json:34]:
```suggestion
    ]
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/gkeyboard/greatpad/info.json (hunk 3 above)
Comment (gkeyboard) [reply keyboards/gkeyboard/greatpad/info.json]: Rotary Encoder  not work if delete "enable":true,

File: keyboards/gkeyboard/greatpad/readme.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+* Hardware Supported: GreatPad PCB
+* Hardware Availability: http://www.mltelectronic.com
+
+Make example for this keyboard (after setting up your build environment):
+
+    make greatpad:default
+
+Flashing example for this keyboard:
+
+    make greatpad:default:flash
```
Comment (fauxpark) [inline keyboards/gkeyboard/greatpad/readme.md:15]:
```suggestion
    make gkeyboard/greatpad:default

Flashing example for this keyboard:

    make gkeyboard/greatpad:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/gkeyboard/greatpad/greatpad.c [hunk 12]
```diff
@@ -0,0 +1,577 @@
...
+static void render_info(void) {
+    oled_write_P(PSTR("       GreatPad\n"), false);
+    oled_write_P(PSTR("\n"), false);
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.caps_lock ? PSTR("Cap(x) ") : PSTR("Cap( ) "), false);
+    oled_write_P(led_state.num_lock ? PSTR("Num(x) ") : PSTR("Num( ) "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("Scrl(x)") : PSTR("Scrl( )"), false);
+    oled_write_P(PSTR("\n"), false);
+
+    switch (get_highest_layer(layer_state)) {
```
[the remark this answers is not in the document]
Comment (gkeyboard) [reply keyboards/gkeyboard/greatpad/greatpad.c:524]: Thanks you.

File: keyboards/gkeyboard/greatpad/info.json
Comment (fauxpark) [inline keyboards/gkeyboard/greatpad/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/gkeyboard/greatpad/info.json
Comment (gkeyboard) [reply keyboards/gkeyboard/greatpad/info.json:1]: ok

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**